hadoop-hdds/docs/content/feature/Snapshot.md

```diff
@@ -23,55 +23,260 @@ summary: Ozone Snapshot
+* **Snapshots During Ongoing Deletes:** If a large recursive delete is executed on a bucket and a snapshot is taken concurrently (or just before the delete), it’s possible that the space will not be reclaimed until the snapshot is removed. In simpler terms, if you delete a large directory (with many keys) from the active bucket but you had created a snapshot right before that delete (or during the deletion process), all those deleted keys remain protected by the snapshot. They will consume space on disk until the snapshot that references them is deleted. This is expected (since snapshots preserve data), but the caveat is that if an admin is not aware a snapshot was taken, it may appear that a delete did not free up space. The space will be freed only after deleting the snapshot and allowing the background cleaner to run. In practice, avoid scheduling snapshots at exactly the same time as massive deletes, or be mindful that snapshots will “lock” data from deletion.
+* **Reserved Namespace Name `.snapshot`:** The name `.snapshot` is reserved by Ozone. Users cannot create keys or directories with the exact name `.snapshot` at the root of a bucket (and similarly, `.snapshot` cannot be used as a volume or bucket name). This is because `.snapshot` is used as the special entry to access snapshots. For example, if a user tries to create a directory literally named `.snapshot` in a bucket, the system will prevent it or treat it as the snapshot namespace. This is analogous to HDFS where `.snapshot` is a reserved directory name in snapshottable directories. The reservation is case-sensitive (only `.snapshot` in all lowercase is reserved). As a best practice, avoid naming any key or folder in Ozone starting with “.snapshot” to prevent any ambiguity.
+* **Snapshot Performance and Scale:** Creating snapshots is instantaneous and cheap in Ozone; however, maintaining a very large number of snapshots (thousands per bucket) can increase the size of OM metadata and slow down operations like listing snapshots or performing snapshot diff (especially if many snapshots are being diffed). Recent improvements have raised the supported snapshot limit (default max 10k per OM as noted above) and optimized diff performance. Nevertheless, if you have an extremely high snapshot count, monitor the OM for any memory or performance impact. Likewise, snapshot diff operations that produce extremely large outputs (e.g., millions of differences) may be resource-intensive to serve; they are paginated and cached to mitigate this.
```

Comment:
Snapshot diff performance doesn't get directly impacted because of the number of snapshots as such. Only the number of concurrent snapshot diff operations will impact snapshot diff performance because of throttling mechanisms in place.